### PR TITLE
[Test]Reduce rabbitmq failure rate

### PR DESCRIPTION
**Pull request by Renkai**
The RabbitMQ service in tests startup asynchronously, so connect sometimes failed because the service is not prepared yet, try multiple times to reduce the failure rate.

### CODE REVIEW

File: pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java [hunk 1]
```diff
@@ -65,7 +64,23 @@ public void TestOpenAndWriteSink() throws Exception {
         RabbitMQSink sink = new RabbitMQSink();
 
         // open should success
-        sink.open(configs, null);
```
Comment (aahmed-se) [inline pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java:68]: Use https://github.com/awaitility/awaitility it's already part of the repo , add a setup method for rabbitmq.

**Review by aahmed-se — CHANGES_REQUESTED** [review]

File: pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java (hunk 1 above)
Comment (Renkai) [reply pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java:68]: @aahmed-se updated, PTAL again.

**Review by aahmed-se — APPROVED** [review]

**Review by jiazhai — APPROVED** [review]

File: pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java [hunk 2]
```diff
@@ -65,7 +66,15 @@ public void TestOpenAndWriteSink() throws Exception {
...
+        // rabbitmq service may need time to initialize
+        Awaitility.await().atMost(Duration.ofSeconds(5)).until(() -> {
+            try {
+                sink.open(configs, null);
+                return true;
```
Comment (lhotari) [inline pulsar-io/rabbitmq/src/test/java/org/apache/pulsar/io/rabbitmq/sink/RabbitMQSinkTest.java:73]: One way to make the code cleaner is to use `.untilAsserted` that can be used for code that isn't expected to throw exceptions.
I'd also recommend omitting `atMost(Duration.ofSeconds(5))` part. The default is 10 seconds by default and that should be fine

**Review by codelipenghui — APPROVED** [review]

**Review by lhotari — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by aahmed-se — APPROVED** [review]

**Review by zymap — APPROVED** [review]

**Pull request merged**